### DISCUSSION TITLE
How to know if RX/TX pins are assigned?

### DISCUSSION THREAD

**Discussion by x-channel** [micropython/micropython#13633] [ESP8266]
Hello everyone, I got little problem with esp8266. When I try to assign GPIO1 or GPIO3, the REPL stop to work and say the device got busy, because this pins run as UART serial. There way to know assigned pins?

**Comment by robert-hh** [comment]
The ESP8266 has just one UART with TX and RX available, and that one is used for the REPL I/O. There is a second UART, but only it's TX pin is available at GPIO2. And different to the ESP32, the assignment of UART signals to pins is fixed. GPIO port 0, 2, 4, 5, 12, 13, 14, 15, 16 are usually available, as well as the ADC Pin for slow input signals.
Edit: If you want to use Pin 1 and 3 for other purposes, you can detach REPL from these pins using os.dupterm() and use WEBREPl for the REPL interface.

**Reply by x-channel** [reply]
Thanks for reply Robert. I really want know how to discovery if device is connected via USB, I already find some people with same concern as me. Then, discovery assigned pins will give much versatility and security to Micropython.

**Reply by robert-hh** [reply]
With the ESP8266 you cannot tell, whether the device is connected to USB, simply because the ESP8266 is NOT connected to USB. It a ESP8266 board has a USB interface, then there is as USB/Serial bridge like a CH340 or CP2102 connecting the USB to ESP8266.
